modal/cli/run.py

```diff
@@ -18,6 +19,9 @@
 
 run_cli = typer.Typer(name="run")
 
+# Used to set Modal cleaner app descriptions which don't include modal CLI parts, only
+# user code parts.
+modal_cli_app_desc: str = None
```

Comment:
A possible alternative to avoid using the global might be to let `_get_click_command_for_function` set the description on the stub (if it isn't already set) before running it? Then we can just pass the "default" description to _get_click_command_for_function below?
It would require moving the `_infer_app_desc` to some later point of execution though (like when accessing .description) instead of running it at stub __init__ like we do now though

Reply:
Thanks for the idea! I'll give it a shot because adding globals is 🤮

Reply:
@freider alright I implemented something along the lines of your suggestion. It's not an equivalent implementation though, because the `global` mechanism allowed the stub to have the correct description as soon as it was imported. See https://github.com/modal-labs/modal-client/pull/348/files#diff-d1f9598cb3b725966536fc17038e0b6c9eaf84409994a573a248b3da948f74caR9.